appserver/payara-appserver-modules/hazelcast-ejb-timer/src/main/java/fish/payara/ejb/timer/hazelcast/HazelcastTimerStore.java

```diff
@@ -96,6 +101,7 @@ static void init(HazelcastCore core) {
         }
     }
 
+    @SuppressWarnings("LeakingThisInConstructor")
```

Comment:
Hmmmm really don't like this. Can the DataGridEjbTimerService register this class when it creates it?

Reply:
@MattGill98 I assume you refer to the `Globals` access and registering this as a cluster listener? Can you motivate why you think this should occur in the `DataGridEjbTimerService` for me? I don't see the connection but I also don't know the details of the timer infrastructure.

Reply:
That's the class that initialises this one 😉